src/ast/ast.h

```diff
@@ -719,6 +691,35 @@ std::string opstr(const Jump &jump);
 
 SizedType ident_to_record(const std::string &ident, int pointer_level = 0);
 
+template <typename T>
+concept NodeType = std::derived_from<T, Node>;
+
+/*
+ * Manages the lifetime of AST nodes.
+ *
+ * Nodes allocated by an ASTContext will be kept alive for the duration of the
+ * owning ASTContext object.
+ */
```

Comment:
Neat idea. Basically an arena, right?

Reply:
Yep, a poor man's arena for now :P 
One day we might want to move it over to a more efficient allocation scheme